snapcraft/plugins/python.py

```diff
@@ -346,7 +351,19 @@ def build(self):
 
         setup_file = os.path.join(self.builddir, 'setup.py')
         with simple_env_bzr(os.path.join(self.installdir, 'bin')):
-            self._run_pip(setup_file)
+            installed_pipy_packages = self._run_pip(setup_file)
+        # We record the requirements and constraints files only if they are
+        # remote. If they are local, they are already tracked with the source.
+        if self.options.requirements and isurl(self.options.requirements):
```

Comment:
this check need improvement as != url does not imply it is part of the source. You can do this in a follow up PR though.